src/track/beatutils.cpp

```diff
@@ -5,44 +5,350 @@
  *      Author: vittorio
  */
 
-#include <algorithm>
-#include <QtDebug>
-#include <QString>
+#include "track/beatutils.h"
+
 #include <QList>
 #include <QMap>
+#include <QString>
+#include <QtDebug>
+#include <algorithm>
 
-#include "track/beatutils.h"
 #include "util/math.h"
+#include "util/descriptivestatistics.h"
+#include "util/windowedstatistics.h"
+
+namespace {
 
 // we are generous and assume the global_BPM to be at most 0.05 BPM far away
 // from the correct one
-#define BPM_ERROR 0.05
+constexpr double kMaxBpmError = 0.05;
+// When ironing the grid for long sequences of const tempo we use
+// a 25ms tolerence since this will sound unnoticed
+constexpr double kMaxSecsPhaseError = 0.025;
 
 // the raw beatgrid is divided into blocks of size N from which the local bpm is
 // computed. Tweaked from 8 to 12 which improves the BPM accuracy for 'problem songs'.
-#define N 12
+constexpr int kBeatsToCountTempo = 12;
 
 static bool sDebug = false;
 
 const double kCorrectBeatLocalBpmEpsilon = 0.05; //0.2;
 const int kHistogramDecimalPlaces = 2;
 const double kHistogramDecimalScale = pow(10.0, kHistogramDecimalPlaces);
 const double kBpmFilterTolerance = 1.0;
+// when comparing with two tempos should be the same we use this tolerence
```

Comment:
```suggestion
// maximum difference for BPMs to be considered the same
```
I'm unclear why both this and kMaxSecsPhaseError exist. Could this be defined in terms of kMaxSecsPhaseError?

Reply:
Interesting idea. Originally these are independent and used in a very different context but I guess it makes sense to be defined in terms of the phase error.